### ISSUE TITLE
Splatting creates spurious warning about extending function without using module defined typed arguments

### ISSUE THREAD

**Issue by CameronBieganek** [julia-vscode/StaticLint.jl#224]
![image](https://user-images.githubusercontent.com/8310743/100407259-87816d80-302d-11eb-977c-82ac75c32a48.png)

![image](https://user-images.githubusercontent.com/8310743/100407285-9cf69780-302d-11eb-8688-45ccefd9fdf8.png)

![image](https://user-images.githubusercontent.com/8310743/100407307-b3045800-302d-11eb-871c-8c67646888b2.png)

**Comment by CameronBieganek** [comment]
Also this:

![image](https://user-images.githubusercontent.com/8310743/100408062-ad0f7680-302f-11eb-83bc-40a3f01b79a9.png)

**Comment by pfitzseb** [comment]
At least the first case is valid, no? You define `Base.cat(us::Union{}; dims::Integer)` there, which looks like piracy to me. 
Dunno where the types come from in the second case, so maybe that actually is a bug.

**Comment by CameronBieganek** [comment]
> You define `Base.cat(us::Union{}; dims::Integer)` there

Hmm, I'm not following you. The signature in that function definition is

```julia
Base.cat(us::UnivariateFiniteArray{S,V,R,P,N}...; dims::Integer) where {S,V,R,P,N}
```

Notice that in the second screen capture, the warning goes away after removing the splatting.

That code is from MLJBase.jl. [Here's a link.](https://github.com/alan-turing-institute/MLJBase.jl/blob/2597c5bd5cdcede211f40af0e766d7c0d632a1fa/src/univariate_finite/arrays.jl#L47)

Here's a reproducible example:

```julia
struct A
    x::Int
end

Base.cat(a::A...; dims::Integer) = a.x + 1
```

(I deleted the second example since I can't reproduce it.)

**Comment by pfitzseb** [comment]
You're defining `Base.cat` for zero arguments as well with that definition:
```
julia> f(xs::Int...) = xs
f (generic function with 1 method)

julia> f()
()

julia> f(1)
(1,)

julia> f(1,2)
(1, 2)

julia> f(xs::Float64...) = "float"
f (generic function with 2 methods)

julia> f()
ERROR: MethodError: f() is ambiguous. Candidates:
  f(xs::Int64...) in Main at REPL[1]:1
  f(xs::Float64...) in Main at REPL[9]:1
Possible fix, define
  f()
```

**Comment by pfitzseb** [comment]
Oh, and the usual fix for this is to define `f(x::Int, xs::Int...)` instead.

**Comment by CameronBieganek** [comment]
Oh, ok, makes sense. I didn't write that code, but I'll make a PR to MLJBase.

Although it does seem a little odd to me that `foo(args...)` also defines the `foo()` case. It seems to me that the costs of including the zero argument case in a vararg definition are greater than the benefits.
